### DISCUSSION TITLE
Defining a list of some type with minimum number of items

### DISCUSSION THREAD

**Discussion by ePak** [cue-lang/cue#1602] [Q&A] [answered]
Is it possible to define a list specifying the items' type but also a minimum number of items? e.g.
```
#Child: {
	name: string
}

#Parent: {
	name: string
	children: [...#Child]
	_#AtLeastOneChild: true & list.MinItems(children, 1)
}

mary: #Child
mary: {
	name: "Mary"
}

ben: #Parent
ben: {
	name: "Ben"
	children: [mary]
}
```
error given:
```
#Parent._#AtLeastOneChild: conflicting values false and true:
    ./test.cue:12:21
    ./test.cue:12:28
```
But the check seems to work if I split the definitions and values into two packages and only eval the package with the values.

I have also tried `children: [...#Child] & list.MinItems(1)` which gives `#Parent.children: invalid value [] (does not satisfy list.MinItems(1)):`

Or have I misunderstood how this works?

Thank you!

**Comment by mpvl** [comment, marked as answer]
The problem here is that the evaluation of the definition `#Parent` fails, not the concrete value. This indeed can be circumvented by putting the definition in the package.

This is not desired behavior though. 

It seems to me that the type of error reported by `list.MinItems` is "fatal" where it should be "incomplete". This would cause this behavior. I'll convert this to an issue.
